server/src/com/cloud/hypervisor/HypervisorGuruBase.java

```diff
@@ -138,7 +143,24 @@ protected VirtualMachineTO toVirtualMachineTO(VirtualMachineProfile vmProfile) {
             if(vm.getType() == VirtualMachine.Type.NetScalerVm) {
                 nicProfile.setBroadcastType(BroadcastDomainType.Native);
             }
-            nics[i++] = toNicTO(nicProfile);
+            NicTO nicTo = toNicTO(nicProfile);
+            final NetworkVO network = _networkDao.findByUuid(nicTo.getNetworkUuid());
+            if (network != null) {
+                final Map<NetworkOffering.Detail, String> details = networkOfferingDetailsDao.getNtwkOffDetails(network.getNetworkOfferingId());
+                if (details != null) {
```

Comment:
What about reducing the cyclomatic complexicity here?
it is a matter of inverting the conditional. Instead of `if (true){doSomething}`, we can do `if(!false){continue} doSomething`
This would enable to remove one if inception

Reply:
We're protecting against NPEs and setting details to a `NicTO` object when those details are available. We cannot simply `continue` as the value is being set at lines 161, 163.

Reply:
You are right about the details, but I think I clicked on the wrong line (sorry for the mistake). I was talking about the ` if (network != null) ` check. if ` if (network == null)` you do not do anything, then it is the same as using a continue. I mean, you do a `nics[i++] = nicTo;`, but that can go inside the condition as well. Duplicated lines are not that great, so let's see what else could be done...

We can do something else, we could extract lines 149-160 to a method; this would improve the readability of the code and enable unit tests and Java docs.

Reply:
The block converts `NicProfile` to `NicTO`, the idea of the changes is to override certain security settings on dvswitch's portgroups if they already don't have the settings, using global settings.

Reply:
Thanks for both the explanations and for the effort!

BTW: your explanation is great to enrich our code base. This explanation would be awesome in a Javadoc, so people in the future can understand why we are doing this without needing to deeply inspect the code.